src/utils/helpers.ts

```diff
@@ -1,39 +1,70 @@
+import { COLOR_STATUS } from './constants'
 import { last, sortBy } from 'lodash'
 import { Prescription } from '../types'
 const tokenLocalStorageKey = 'token'
 
-const getTokenLocalStorage = (): (string|null) => localStorage.getItem(tokenLocalStorageKey)
+const getTokenLocalStorage = (): string | null =>
+  localStorage.getItem(tokenLocalStorageKey)
 
-const setTokenLocalStorage = (token: string): void => localStorage.setItem(tokenLocalStorageKey, token)
+const setTokenLocalStorage = (token: string): void =>
+  localStorage.setItem(tokenLocalStorageKey, token)
 
-const removeTokenLocalStorage = (): void => localStorage.removeItem(tokenLocalStorageKey)
+const removeTokenLocalStorage = (): void =>
+  localStorage.removeItem(tokenLocalStorageKey)
 
 /**
  * Returns the first valid prescription found or null
  */
-const getValidPrescription = (prescriptions: Prescription[]): Prescription|null => (
+const getValidPrescription = (
+  prescriptions: Prescription[]
+): Prescription | null =>
   prescriptions.filter((prescription) => prescription.is_valid)[0] ?? null
-)
 
 /**
  * Returns the last prescription (by `end_date`) or null if none.
  */
-const getLastPrescription = (prescriptions: Prescription[]): Prescription|null => (
-  last(sortBy(prescriptions, 'end_date')) ?? null
-)
+const getLastPrescription = (
+  prescriptions: Prescription[]
+): Prescription | null => last(sortBy(prescriptions, 'end_date')) ?? null
 
 /**
  * Returns a valid prescription if found or default to the last one
  */
-const getValidOrLastPrescription = (prescriptions: Prescription[]): Prescription|null => (
+const getValidOrLastPrescription = (
+  prescriptions: Prescription[]
+): Prescription | null =>
   getValidPrescription(prescriptions) ?? getLastPrescription(prescriptions)
-)
+
+/**
+ * This function returns a color based on the validity of a prescription and if it expires soon
+ * @param prescription - The prescription object
+ * @param patient - The patient object
+ * @returns The color of the badge
+ */
+const getIconClass = (prescription: Prescription): string => {
+  const statusMapping = {
+    expiring_soon: COLOR_STATUS.WARNING,
+    is_valid: COLOR_STATUS.SUCCESS,
+    default: COLOR_STATUS.DANGER
+  }
+
+  if (prescription.expiring_soon) {
+    return statusMapping.expiring_soon
+  }
+
+  if (prescription.is_valid) {
+    return statusMapping.is_valid
+  }
+
+  return statusMapping.default
+}
```

Comment:
We're almost there, but please unit test that function you will see it has a few bugs

Reply:
I write a unit test